internal/server/server.go

```diff
@@ -551,6 +553,35 @@ func (s *Server) VerifyIdentityHeader(nextHandler echo.HandlerFunc) echo.Handler
 	return func(ctx echo.Context) error {
 		request := ctx.Request()
 
+		loader := openapi3.NewSwaggerLoader()
+		doc, err := loader.LoadSwaggerFromFile("api.yaml")
```

Comment:
This won't actually work since api.yaml will not be in the same directory as the binary, in fact it's not really shipped at all.

What I recommend is doing the same thing `GetSwagger` is doing once when setting up the server (in `NewServer`), and the section here until `NewRouter`. And then assigning that router as a private field in the `Server` struct.